### PR TITLE
Start updating distributing-packages-using-setuptools guide

### PR DESCRIPTION
**Pull request by jeanas**
The [Packaging and Distributing Projects](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/) guide is quite outdated. It uses `setup.py` and no `pyproject.toml`. It's also out of sync with the sample project.

I'd like to update it, but this is going to be difficult to review if I do it in a single sweeping PR, so here's a start, to be continued later. This PR removes the descriptions of `setup()` arguments that have better equivalents in `[project]`. Where these descriptions had more details than the [Writing pyproject.toml](https://packaging.python.org/guides/writing-pyproject-toml/) guide, I have moved the details there.

----
📚 Documentation preview 📚: https://python-packaging-user-guide--1450.org.readthedocs.build/en/1450/

### CODE REVIEW

File: source/guides/distributing-packages-using-setuptools.rst [hunk 1]
```diff
@@ -4,6 +4,10 @@
 Packaging and distributing projects
 ===================================
 
+.. warning::
+   As of December 2023, this page is largely outdated
+   and currently being revised.
+
```
Comment (chrysle) [inline source/guides/distributing-packages-using-setuptools.rst:10]:
```suggestion
:Page Status: Outdated
:Last Reviewed: 2023-12-14
```

**Review by chrysle — COMMENTED** [review]
Thanks for doing this!

File: source/guides/distributing-packages-using-setuptools.rst [hunk 2]
```diff
@@ -4,6 +4,8 @@
 Packaging and distributing projects
 ===================================
 
+:Page Status: Outdated
```
Comment (jeanas) [inline source/guides/distributing-packages-using-setuptools.rst:7]:
```suggestion
:Page Status: Outdated

```

File: source/guides/distributing-packages-using-setuptools.rst [hunk 3]
```diff
@@ -4,6 +4,9 @@
...
+
+:Last Reviewed: 2023-12-14
```
Comment (chrysle) [inline source/guides/distributing-packages-using-setuptools.rst:9]:
```suggestion
:Last Reviewed: 2023-12-14

```

Sorry, missed this.

**Review by chrysle — APPROVED** [review]

**Pull request merged**